tests/e2e/migrations_test.go

```diff
@@ -99,6 +100,56 @@ func TestMigrateUpTo(t *testing.T) {
 	check.Number(t, gotVersion, upToVersion) // incorrect database version
 }
 
+func TestMigrateUpTimeout(t *testing.T) {
```

Comment:
I think these tests are simulating the library timeout, but we should probably have something that exercises the `--timeout` flag and uses `pg_sleep` (sqlite would be ideal, but I don't think there's a way to do that in sqlite)

Reply:
Made some changes in https://github.com/pressly/goose/pull/627/commits/303331cbbd96f652f4039f2c2cc568e3e108de3b, I think this captures what you described above, and exercises a "long running" migration instead of starting with an already cancelled context.